fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/QuantifiedApply2JoinRule.java

```diff
@@ -51,27 +53,51 @@ public boolean check(OptExpression input, OptimizerContext context) {
                 && !SubqueryUtils.containsCorrelationSubquery(input);
     }
 
+    private static ScalarOperator normalizeBinaryOperator(BinaryPredicateOperator input) {
+        if (input.getUsedColumns().isEmpty()) {
+            return new FoldConstantsRule().visitBinaryPredicate(input, null);
+        } else if (input.getChild(0).getUsedColumns().isEmpty()) {
+            // normalize binaryPredicateOperator, e.g. 1 = tbl.v1 to tbl.v1 = 1
+            input.swap();
```

Comment:
You can do this with NormalizePredicateRule

Reply:
subquery maybe can't do that now

Reply:
you can do this with ScalarOperatorRewriter's DEFAULT_REWRITE_RULES

Reply:
done, replaced with DEFAULT_REWRITE_RULES